tag_release.py

```diff
@@ -0,0 +1,554 @@
+        auth=(username, password),
+        headers={"User-Agent": TOKEN_NAME},
+    )
+    if not response.ok:
+        message = response.json()["message"]
+        raise requests.exceptions.RequestException(message)
+
+
+def ensure_github_creds(attempts=3):
+    """
+    Make sure that we have GitHub OAuth credentials. This will check the user's
+    .netrc file, as well as the ~/.config/openedx-release file. If no credentials
+    exist in either place, it will prompt the user to create OAuth credentials,
+    and store them in ~/.config/openedx-release.
+
+    Returns False if we found credentials, True if we had to create them.
+    """
+    if get_github_creds():
+        return False
+
+    # Looks like we need to create the OAuth creds
+    print("We need to set up OAuth authentication with GitHub's API. "
+          "Your password will not be stored.", file=sys.stderr)
+    token = None
+    for _ in range(attempts):
+        try:
+            username, token = create_github_creds()
+        except requests.exceptions.RequestException as e:
+            print(
+                "Invalid authentication: {}".format(e.message),
+                file=sys.stderr,
+            )
+            continue
+        else:
+            break
+    if token:
+        print("Successfully authenticated to GitHub", file=sys.stderr)
+    if not token:
+        print("Too many invalid authentication attempts.", file=sys.stderr)
+        return False
+
+    config_file = (path("~/.config") / TOKEN_NAME).expand()
+    # make sure parent directory exists
+    config_file.parent.makedirs_p()
+    # read existing config if it exists
+    if config_file.isfile():
+        with open(config_file) as f:
+            config = json.load(f)
+    else:
+        config = {}
+    # update config
+    if 'credentials' not in config:
+        config["credentials"] = {}
+    if 'api.github.com' not in config['credentials']:
+        config["credentials"]["api.github.com"] = {}
+    config["credentials"]["api.github.com"]["username"] = username
+    config["credentials"]["api.github.com"]["token"] = token
+    # write it back out
+    with open(config_file, "w") as f:
+        json.dump(config, f)
+
+    return True
+
+def openedx_release_repos(session):
+    repos_resp = session.get(REPOS_YAML)
+    repos_resp.raise_for_status()
+    all_repos = yaml.safe_load(repos_resp.text)
+    repos = {name: data for name, data in all_repos.items()
+             if data and data.get("openedx-release")}
+    return repos
+
+
+def commits_to_tag_in_repos(repos, session):
+    to_tag = {}
+    for repo_name, repo_data in repos.items():
+        # make sure the repo exists
+        repo_url = "https://api.github.com/repos/{repo}".format(repo=repo_name)
+        repo_resp = session.get(repo_url)
+        if not repo_resp.ok:
+            print("Invalid repo {repo}".format(repo=repo_name), file=sys.stderr)
+            continue
```

Comment:
Also, it would be good for debugging to include something from the response in the output.